### DISCUSSION TITLE
Compare local repo to GitHub repo

### DISCUSSION THREAD

**Discussion by w8HAQRHkTx7r** [community/community#22797] [New to GitHub]
<p>I edited a file on my GitHub repo (origin/master) and committed it.  I want to be able to compare my local repo to GitHub.  I expect to see that one (or more) of the files on GitHub are different from what I have locally.</p>
<p>On the local:</p>
<pre><code class="lang-auto">$ git remote -v
origin	https://github.com/&lt;account&gt;/&lt;repo&gt; (fetch)
origin	https://github.com/&lt;account&gt;/&lt;repo&gt; (push)
</code></pre>
<p>but</p>
<pre><code class="lang-auto">$ git log --oneline --decorate --graph --all
* 42f3624 (HEAD -&gt; master, origin/master) print usage def
* d6d0d0e bad file name bug fixed
</code></pre>
<p>is not showing me the commit that was made on GitHub.  I’m expecting to see something about commit ef714fa</p>
<pre><code class="lang-auto">Commits on Oct 4, 2020
Added less -R  …
@&lt;account&gt; committed 41 minutes ago Verified  ef714fa  

Commits on Sep 25, 2020
print usage def
@&lt;account&gt; committed 9 days ago 42f3624  
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>Did you do a <code>git fetch origin</code> to get updates from the repository on Github? <code>git log</code> operates only on the locally known data.</p>

**Comment by w8HAQRHkTx7r** [comment]
<p>I had not done a fetch.  As I read about that command, it makes sense.  I guess that when I saw a reference to origin/master in the 42f36 commit (see below), it was telling me how things are right now.</p>
<pre><code class="lang-auto">$ git log -3
commit 42f36242037e8d7652f151785cdac618da738e0d (HEAD -&gt; master, origin/master)
Author: MACE &lt;maceww6438@gmail.com&gt;
Date:   Fri Sep 25 11:58:14 2020 +0000

    print usage def
</code></pre>
<p>But I guess it shows how things were the last time the local repo looked at the remote repo.  That “look” happened on my last push.  And I’ll need something else to cause a “look” – like a fetch.</p>
<p>Thanks for the enlightenment!</p>
